diff --git a/src/shaders/shapecorners.glsl b/src/shaders/shapecorners.glsl
index 8ec4724..5f43bce 100644
--- a/src/shaders/shapecorners.glsl
+++ b/src/shaders/shapecorners.glsl
@@ -22,9 +22,7 @@ vec2 pixel_to_tex(vec2 pixelcoord) {
 }
 bool isDrawingShadows() { return  windowSize != windowExpandedSize && shadowColor.a > 0.0; }
 bool isDrawingOutline() {
-    vec2 one_edge = vec2(windowSize.x/2.0, 0.0);
-    return  texture2D(sampler, pixel_to_tex(one_edge)).a > 0.5 &&
-    outlineColor.a > 0.0 && outlineThickness > 0.0;
+    return outlineColor.a > 0.0 && outlineThickness > 0.0;
 }
 
 float parametricBlend(float t) {
