diff --git a/src/shaders/shapecorners.glsl b/src/shaders/shapecorners.glsl
index 8ec4724..c0eb420 100644
--- a/src/shaders/shapecorners.glsl
+++ b/src/shaders/shapecorners.glsl
@@ -76,7 +76,7 @@ vec4 shapeCorner(vec2 coord0, vec4 tex, vec2 start, float angle) {
     }
 
     if(isDrawingOutline()) {
-        vec4 outlineOverlay = vec4(mix(tex.rgb, outlineColor.rgb, outlineColor.a), tex.a);
+        vec4 outlineOverlay = vec4(mix(tex.rgb, outlineColor.rgb, outlineColor.a), 1.0);
 
         if (distance_from_center < r - outlineThickness + 0.5) {
             // from the window to the outline
